cli/deployment/config.go

```diff
@@ -41,17 +48,59 @@ func NewConfig(data string, file string) Config {
 		data: data,
 	}
 
+	if len(file) > 0 {
+		config.file = filepath.Clean(file)
+		config.dir = filepath.Dir(file)
+	} else {
+		// yaml passed as sting through parameter
+		dir, err := os.Getwd()
+		if err != nil {
+			log.Fatalf("could not get current folder path: %v", err)
+		}
+		config.dir = dir
+	}
+
 	err := yaml.Unmarshal([]byte(data), &config)
 	if err != nil {
-		log.Fatalf("error: %v", err)
+		log.Fatalf("failed unmarshal config yaml %s, error: %v", file, err)
+	}
+	// check project id and deployment name
+	if len(config.Project) == 0 {
+		config.Project = DefaultProjectID
+	}
+	if len(config.Name) == 0 {
+		if len(config.file) == 0 {
+			// if file empty, than yaml string as a parameter was passed, it SHOULD contain name field
+			log.Fatalf("name field not defined in yaml: %s", data)
+		} else {
+			config.Name = DeploymentNameFromFile(config.file)
+		}
 	}
 	return config
 }
 
+// GetProject returns project id if it set in config file, otherwise default value defined in DefaultProjectID var.
+func (c Config) GetProject() string {
+	if len(c.Project) > 0 {
+		return c.Project
+	} else {
+		return DefaultProjectID
+	}
+}
+
+// returns file path in case of file source or yaml string in case of yaml source
```

Comment:
Per https://golang.org/doc/effective_go.html#commentary this should read:

```suggestion
// Source returns the the path of the file for a file-backed configuration, or a YAML string for YAML configurations.
```